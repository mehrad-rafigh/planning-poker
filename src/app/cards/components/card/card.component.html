<div
  class="card"
  [ngStyle]="{ 'background-color': cardColor.backgroundColor }"
  [ngClass]="setFontColor()"
  (click)="handleClick()"
>
  <span *ngIf="!detailMode" class="top-right quarter-h">{{ value }}</span>
  <span *ngIf="detailMode" class="half-h" [ngClass]="setDetailMode()">{{
    value
  }}</span>
  <span *ngIf="!detailMode" class="bottom-left quarter-h">{{ value }}</span>
</div>
