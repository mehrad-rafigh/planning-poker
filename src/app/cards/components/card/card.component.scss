@import "../../../../mixins";

span {
  font-weight: bold;
}

.card {
  display: flex;
  flex-flow: column nowrap;
  box-sizing: border-box;
  height: 200px;
  width: auto;
  border-color: lightslategray;
  position: relative;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  border-radius: 15px;

  @include mobile() {
    height: 250px;
    font-size: 26px;
  }

  @include tablet() {
    height: 300px;
    font-size: 24px;
  }

  @include above-tablet() {
    height: 350px;
    font-size: 22px;
  }
}

.top-right {
  text-align: right;
  margin-top: 32px;
  margin-right: 32px;
}

.bottom-left {
  margin-bottom: 32px;
  margin-left: 32px;
  position: absolute;
  bottom: 0;
  height: auto !important;
}

.quarter-h {
  height: 25%;
}

.half-h {
  height: 50%;
}

.detail-mode {
  height: unset;
  margin: auto;
  font-size: 128px;
}

.font-white {
  color: white;
}
