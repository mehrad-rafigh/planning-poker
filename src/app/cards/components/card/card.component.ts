import {
  Component,
  OnInit,
  ChangeDetectionStrategy,
  Input,
  EventEmitter,
  Output
} from "@angular/core";

@Component({
  selector: "app-card",
  templateUrl: "./card.component.html",
  styleUrls: ["./card.component.scss"],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class CardComponent implements OnInit {
  @Input() value: string;
  @Input() cardColor: { backgroundColor: string; fontColor: string };
  @Input() detailMode = false;
  @Output() clickEvent = new EventEmitter<any>();
  constructor() {}

  ngOnInit(): void {}

  handleClick(): void {
    this.clickEvent.emit({
      value: this.value,
      card: {
        backgroundColor: this.cardColor.backgroundColor,
        fontColor: this.cardColor.fontColor
      }
    });
  }

  setDetailMode(): string {
    if (this.detailMode) {
      return "detail-mode";
    }
  }

  setFontColor(): string {
    if (this.cardColor.fontColor === "white") {
      return "font-white";
    }
  }
}
