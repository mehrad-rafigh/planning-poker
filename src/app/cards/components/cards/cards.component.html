<app-toolbar></app-toolbar>
<div class="cards-container">
  <app-card
    *ngFor="let no of range"
    [value]="no"
    [cardColor]="setCardColors()"
    (clickEvent)="showCardInDetail($event)"
  ></app-card>
</div>
