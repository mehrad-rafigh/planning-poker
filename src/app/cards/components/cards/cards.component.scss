@import "../../../../mixins";

.cards-container {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, 1fr);
  grid-gap: 16px;
  margin: 16px;

  @include mobile() {
    grid-template-columns: repeat(2, 1fr);
  }

  @include tablet() {
    grid-template-columns: repeat(3, 1fr);
  }

  @include above-tablet() {
    grid-template-columns: repeat(4, 1fr);
  }
}
