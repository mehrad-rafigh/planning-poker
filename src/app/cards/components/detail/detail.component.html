<app-toolbar></app-toolbar>
<app-card
  [detailMode]="true"
  [value]="value"
  [cardColor]="card"
></app-card>
