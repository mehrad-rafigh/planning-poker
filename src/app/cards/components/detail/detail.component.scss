@import "../../../../mixins";

app-card ::ng-deep div {

  margin: 0 auto;



  @include mobile() {
    width: 290px !important;
    height: 450px !important;
    margin-top: 32px;
  }

  @include tablet() {
    width: 350px !important;
    height: 500px !important;
    margin-top: 48px;
  }

  @include above-tablet() {
    width: 400px !important;
    height: 550px !important;
    margin-top: 72px;
  }


}
