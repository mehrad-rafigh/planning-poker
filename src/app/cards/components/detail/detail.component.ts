import {
  Component,
  OnInit,
  ChangeDetectionStrategy,
  Input
} from "@angular/core";
import { Router } from "@angular/router";

@Component({
  selector: "app-detail",
  templateUrl: "./detail.component.html",
  styleUrls: ["./detail.component.scss"],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class DetailComponent implements OnInit {
  @Input() value: string;
  @Input() card: { backgroundColor: string; fontColor: string };
  constructor(private router: Router) {
    const state = this.router.getCurrentNavigation().extras.state;

    if (!state) {
      this.router.navigate(["/"]);
    }

    this.card = state.card;
    this.value = state.value;
  }

  ngOnInit() {}
}
