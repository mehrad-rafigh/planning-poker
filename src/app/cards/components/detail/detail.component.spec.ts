import { async, ComponentFixture, TestBed } from "@angular/core/testing";

import { DetailComponent } from "./detail.component";
import { ToolbarComponent } from "../../../toolbar/components/toolbar/toolbar.component";
import { CardComponent } from "../card/card.component";
import { RouterTestingModule } from "@angular/router/testing";
import { CardsComponent } from "../cards/cards.component";
import { Router } from "@angular/router";

describe("DetailComponent", () => {
  let component: DetailComponent;
  let fixture: ComponentFixture<DetailComponent>;
  let router: Router;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [
        DetailComponent,
        ToolbarComponent,
        CardComponent,
        CardsComponent
      ],
      providers: [
        {
          provide: Router,
          useClass: class RouterSub {
            getCurrentNavigation() {
              return {
                extras: {
                  state: {
                    card: {
                      backgroundColor: "rgb(100,100,100)",
                      fontColor: "white"
                    },
                    value: ""
                  }
                }
              };
            }
          }
        }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DetailComponent);
    component = fixture.componentInstance;
    router = TestBed.inject(Router);

    component.card = {
      backgroundColor: "rgb(100,100,100)",
      fontColor: "white"
    };
    component.value = "0.5";

    fixture.detectChanges();
  });

  it("should create", () => {
    expect(component).toBeTruthy();
  });

  it("should match the snapshot", () => {
    expect(fixture).toMatchSnapshot();
  });
});
