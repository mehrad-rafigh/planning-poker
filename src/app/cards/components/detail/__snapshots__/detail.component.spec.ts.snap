// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailComponent should match the snapshot 1`] = `
<app-detail
  card={[Function Object]}
  router={[Function RouterSub]}
  value={[Function String]}
>
  <app-toolbar>
    <nav
      class="navbar d-flex row-no-wrap"
    >
      <span
        class="navbar-title f-start"
      >
        Planning Poker
      </span>
    </nav>
  </app-toolbar><app-card>
    <div
      class="card font-white"
      style="background-color: rgb(100, 100, 100);"
    >
      
      
      <span
        class="half-h detail-mode"
      >
        0.5
      </span>
      
    </div>
  </app-card>
</app-detail>
`;
